### PR TITLE
Added "multiple instances of this field" functionality

### PR DESCRIPTION
**Pull request by Pomax**
fixes https://github.com/Pomax/react-formbuilder/issues/10

Currently supported: `input type=text`

Run `npm start` and then open the `./example/index.html` file in your browser: the "participant name" should be a multiplicity field, set to 1 by default but can be given more/fewer with (+) and (-) buttons.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,9 +42,11 @@ Where a field definition object takes the following form:
   placeholder: String data (optional)
   metered: boolean (optional),
   optional: boolean (optional),
+  options: array of strings for each option when "type" is choiceGroup, checkboxGroup, or a React component that takes "options" (like a ReactSelect component),
```
Comment (mmmavis) [inline README.md]: this is a @mmmavis problem 😉 I'm trying to understand what `options` is for here. Is each `option` string here the label text we show on the UI?

Comment (Pomax) [inline README.md]: yeah it's the various options that you can select. If it's a radio group, you can only pick one, if it's checkboxes, then you can select multiple, and if it's something like a pull down menu, think `<select>` options.

So you can specify something like:
```
  type: "checkboxGroup",
  options: ["toddler", "preteen", "teen", "young adult", "adult"],
  ...
```
and then the UI would show something like

- [ ] toddler
- [ ] toddler
- [ ] toddler

File: src/MultiplicityField.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    inputClass = (inputClass + " multiple").trim();
+
+    if (ftype === "undefined" || Type === "text") {
+      formfield = <input className={inputClass} type={Type? Type : "text"} {...common}/>;
+    } else if (Type === "textarea") {
+      formfield = <textarea className={inputClass} {...common}/>;
+    }
+
+    var removeButton = position > 0 ? <button type="button" onClick={() => this.removeField(position)}>(-)</button> : null;
```
Comment (Pomax) [inline src/MultiplicityField.jsx]: not sure what to use as defaults here - maybe a `{ this.props.fewerLabel || "(-)" } and corresponding `moreLabel`?

**Comment by Pomax** [comment]
Good finds, I'll fix both issues.

**Comment by Pomax** [comment]
PR incoming after a rebase. Also you know how you make "textarea" work? You uncomment the code for it... yeeees.... >_>;;

**Comment by Pomax** [comment]
@alanmoo re-R?

**Comment by Pomax** [comment]
I see a test

**Pull request merged**